File: 9012.py
import sys

n = int(sys.stdin.readline())


def vps(li):
    stack = []

    if li[0] == ")":
        return "NO"
    else:
        for s in li:
            if s == "(":
                stack.append(s)
            elif s == ")" and stack != []:
                stack.pop()
            else:
                return "NO"

        if stack == []:
            return "YES"
        else: return "NO"

for i in range(n):
    order = list(sys.stdin.readline().rstrip())
    print(vps(order))

File: Go_solutions/9012.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func vps(x string) string {
	var stack []string

	for idx, ch := range x {
		s := string(ch)
		if idx == 0 && s == ")" {
			return "NO"
		} else {
			if s == "(" {
				stack = append(stack, s)
			} else if s == ")" && stack != nil {
				length := len(stack)
				if length == 1 {
					stack = nil
				} else {
					stack = stack[:length-1]
				}
			} else {
				return "NO"
			}
		}
	}
	if stack == nil {
		return "YES"
	} else {
		return "NO"
	}
}

func main() {
	var n int
	inputReader := bufio.NewReader(os.Stdin)
	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		input, _ := inputReader.ReadString('\n')
		action := strings.TrimSpace(input)

		answer := vps(action)

		fmt.Println(answer)
	}
}
